packages/profiler-lib/src/profiler.ts

```diff
@@ -0,0 +1,196 @@
+// Core profiler visualization library
+// This module provides the main API for rendering circuit profiles
+
+import { CircuitProfile } from "./profile.js";
+import { Cytograph, CytographRendering } from "./cytograph.js";
+import { CircuitSelector } from "./selection.js";
+import { MetadataSelector } from './metadataSelection.js';
+
+export interface ProfilerConfig {
+    /** Container element for the graph visualization */
+    graphContainer: HTMLElement;
+    /** Container element for the selector UI controls */
+    selectorContainer: HTMLElement;
+    /** Container element for the navigator minimap */
+    navigatorContainer: HTMLElement;
+    /** Optional container element for the tooltip (defaults to document.body if not provided) */
+    tooltipContainer?: HTMLElement | undefined;
+    /** Optional error message display element */
+    errorContainer?: HTMLElement | undefined;
+    /** Optional message display element for status messages */
+    messageContainer?: HTMLElement | undefined;
+    /** Optional search input element for node search */
+    searchInput?: HTMLInputElement | undefined;
+}
+
+/**
+ * Main profiler class that orchestrates the visualization of circuit profiles.
+ * This is the primary API for embedding the profiler in other applications.
+ */
+export class Profiler {
```

Comment:
I think profiler is the tool which captures the profile
this is a profile visuzalier